File: app/src/models/Penalty.ts
import type { GameId, PenaltyTypeId, TeamId, UserId } from './ids'
import type { DbModel, User } from './index'

export interface PenaltyType extends DbModel {
  name: string
  player_id: UserId
  player: User
  duration: number
  severity: string
}

export interface Penalty extends DbModel {
  player_id: UserId
  team_id: TeamId
  game_id: GameId
  period: number
  game_time: number
  created_by: UserId
  penalty_type: PenaltyType
  penalty_type_id: PenaltyTypeId
}

export type { PenaltyTypeId } from './ids'

File: internal/models/penalty.go
package models

type PenaltyType struct {
	DbModel
	Name     string `json:"name"`
	Duration uint   `json:"duration"`
	Severity string `json:"severity"`
}

type Penalty struct {
	DbModel
	PlayerID      uint        `json:"player_id"`
	TeamID        uint        `json:"team_id"`
	GameID        uint        `json:"game_id"`
	Period        uint        `json:"period"`
	Duration      uint        `json:"duration"`
	CreatedBy     uint        `json:"created_by"`
	PenaltyType   PenaltyType `json:"penalty_type"`
	PenaltyTypeID uint        `json:"penalty_type_id"`
}
